lighthouse-core/audits/third-party-facades.js

```diff
@@ -0,0 +1,214 @@
+
+      if (facadableProductMap.has(product.name)) continue;
+      facadableProductMap.set(product.name, {product, entity});
+    }
+
+    return Array.from(facadableProductMap.values());
+  }
+
+  /**
+   * @param {LH.Artifacts} artifacts
+   * @param {LH.Audit.Context} context
+   * @return {Promise<LH.Audit.Product>}
+   */
+  static async audit(artifacts, context) {
+    const settings = context.settings;
+    const trace = artifacts.traces[Audit.DEFAULT_PASS];
+    const devtoolsLog = artifacts.devtoolsLogs[Audit.DEFAULT_PASS];
+    const networkRecords = await NetworkRecords.request(devtoolsLog, context);
+    const mainResource = await MainResource.request({devtoolsLog, URL: artifacts.URL}, context);
+    const mainEntity = thirdPartyWeb.getEntity(mainResource.url);
+    const tasks = await MainThreadTasks.request(trace, context);
+    const multiplier = settings.throttlingMethod === 'simulate' ?
+      settings.throttling.cpuSlowdownMultiplier : 1;
+    const summaries = ThirdPartySummary.getSummaries(networkRecords, tasks, multiplier);
+    const facadableProducts
+      = ThirdPartyFacades.getFacadableProducts(summaries.byURL, mainEntity);
+
+    /** @type {LH.Audit.Details.TableItem[]} */
+    const results = [];
+    for (const {product, entity} of facadableProducts) {
+      const categoryTemplate = CATEGORY_UI_MAP[product.categories[0]];
+
+      let productWithCategory;
+      if (categoryTemplate) {
+        // Display product name with category next to it in the same column
+        productWithCategory = str_(categoryTemplate, {productName: product.name});
+      } else {
+        // Just display product name if no category is found
+        productWithCategory = product.name;
+      }
+
+      const urls = summaries.urls.get(entity);
+      const entitySummary = summaries.byEntity.get(entity);
+      if (!urls || !entitySummary) continue;
+
+      const items = Array.from(urls)
+        .map((url) => {
+          const urlStats = summaries.byURL.get(url);
+          return /** @type {ThirdPartySummary.URLSummary} */ ({url, ...urlStats});
+        })
+        .sort((a, b) => b.transferSize - a.transferSize);
+      this.condenseItems(items);
+      results.push({
+        product: productWithCategory,
+        transferSize: entitySummary.transferSize,
+        blockingTime: entitySummary.blockingTime,
+        subItems: {type: 'subitems', items},
+      });
+    }
+
+    if (!results.length) {
+      return {
+        score: 1,
+        notApplicable: true,
+      };
+    }
+
+    /** @type {LH.Audit.Details.Table['headings']} */
+    const headings = [
+      /* eslint-disable max-len */
+      {key: 'product', itemType: 'text', subItemsHeading: {key: 'url', itemType: 'url'}, text: str_(UIStrings.columnProduct)},
+      {key: 'transferSize', granularity: 1, itemType: 'bytes', subItemsHeading: {key: 'transferSize'}, text: str_(i18n.UIStrings.columnTransferSize)},
```

Comment:
super nit: use order `key, itemType, subItemsHeading, granularity, text`. order is by usefulness